src/app/heroes/heroes.component.spec.ts

```diff
@@ -0,0 +1,62 @@
+import { HeroesComponent } from './heroes.component';
+import { Hero } from '../hero';
+import { HeroService } from '../hero.service';
+import { of } from 'rxjs';
+
+describe('Heroes Component', ()=> {
+
+  let HEROES : Hero[] = [
+    {id: 1, name: 'Spiderman', strength: 15},
+    {id: 2, name: 'Batman', strength: 10},
+    {id: 3, name: 'Deadpool', strength: 3},
+  ];
+
+  // let mockHeroSvc : HeroService;
+  // let heroesCmp: HeroesComponent;
+  /*
+    * work with above his test is work
+    * in the last lesson became as below
+    * No way pass with me
+  */
+
+  let mockHeroSvc ;
+  let heroesCmp ;
+
+  beforeEach(()=> {
+    // #question how to fix this error
+    // mockHeroSvc = {
+    //   deleteHero: function() {
+    //     return { subscribe: function() {}}
+    //   }
+    // };
+
+    mockHeroSvc = jasmine.createSpyObj<HeroService>(['deleteHero']);
```

Comment:
i believe you are following a course that uses an older version of angular => jasmine

Here is how you create a spy: 

`mockHeroSvc = jasmine.createSpyObj('HeroService', ['deleteHero']);`

![image](https://user-images.githubusercontent.com/35868173/110378657-83b6ab80-805e-11eb-9de5-60ce51e16063.png)

first arg is service name, second is methods, the third one is service properties if it has any